### PR TITLE
Updated the spec to clearly define usage of the sourcemap comment formats

### PR DESCRIPTION
**Pull request by bomsy**
For [#54](https://github.com/tc39/source-map-rfc/issues/54). Updated the supported comments based of the decisions at the last meeting.

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Thanks!

Regarding the failing IPR check: ideally you should be onboarded as a Mozilla delegate so that it's explicit that your contribution are on behalf of your company (cc @dminor could it be done? 🙂), but if it will be a blocker for merging this change we can temporary add an exception for your username.

File: source-map.bs [hunk 1]
```diff
@@ -357,8 +357,8 @@ The generated code should include a line at the end of the source, with the foll
 ```
 
 Note: The prefix for this annotation was initially `//@` however this conflicts with Internet
-Explorer's Conditional Compilation and was changed to `//#`.  It is reasonable for tools to
-also accept `//@` but `//#` is preferred.
+Explorer's Conditional Compilation and was changed to `//#`. Source map generators should only support `//#` 
+while source map consumers should support both `//@` and `//#` for legacy compatibility.
```
Comment (nicolo-ribaudo) [inline source-map.bs:361]:
```suggestion
Explorer's Conditional Compilation and was changed to `//#`. Source map generators must
only emit `//#`, while source map consumers must accept both `//@` and `//#`.
```

_must_, not _should_, as the spec should be as assertive as possible :)

Also, bikeshed supports notes that get formatted as such. They are usually used to describe the reason for something or to give additional information that is not a requirement for implementations. I think we could write the legacy reason in its own paragraph as ``Note: `//@` is needed for compatibility with some existing legacy source maps.``

Comment (bomsy) [reply source-map.bs]: Thanks. I've updated the comments.

**Comment by bomsy** [comment]
> Thanks!
> 
> Regarding the failing IPR check: ideally you should be onboarded as a Mozilla delegate so that it's explicit that your contribution are on behalf of your company (cc @dminor could it be done? 🙂), but if it will be a blocker for merging this change we can temporary add an exception for your username.

What do i need to do to onboard?

**Comment by nicolo-ribaudo** [comment]
@bomsy is now a TC39 delegate, so IPR is good :)

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**